feat: add a processor to ingest to new eap table

kylemumma opened the pull request:
A new EAP clickhouse table was recently created https://github.com/getsentry/snuba/pull/6850
This PR creates a new processor aka kafka consumer that reads spans off the existing `snuba-spans` topic and writes them to the new table.

phacops commented on rust_snuba/src/processors/eap_items.rs:
```diff
@@ -0,0 +1,333 @@
+
+macro_rules! seq_attrs {
+    ($($tt:tt)*) => {
+        seq!(N in 0..20 {
```
We have 40 buckets on the new table.

phacops reviewed (changes requested):
We also need to write the duration, the end timestamp and top-level fields from the message into the attribute buckets.

Also, if we have integers, we double-write them to the `attributes_int`, and booleans into `attributes_bool` (there are not buckets for those columns).

The idea is we want to preserve the original type of the attribute while allowing to aggregate them all at once. We will use the `float` columns to run aggregation or `string` for group bys, and when we need the payload as is, we fetch `attributes_int` and `attributes_bool` and those columns take precedence to represent the real type of the attributes.

kylemumma commented on rust_snuba/src/processors/eap_items.rs:
```diff
@@ -0,0 +1,333 @@
+    pub organization_id: u64,
+    pub project_id: u64,
+    pub item_type: u8,
+    pub timestamp: u64,
```
ok, not sure why eap_spans is doing this stuff https://github.com/getsentry/snuba/blob/2dea118979c06173b186a0917334a56435a35be9/rust_snuba/src/processors/eap_spans.rs#L181

ill just take `timestamp` directly from `start_timestamp` as a `u32`

kylemumma replied:
in `FromSpanMessage` the timestamp is also defined as `u64` so it seems safer to me to leave it as `u64` rather than `u32`. Are you saying that when we store in clickhouse its `DateTime` type which can only hold up to `u32`? In that case do we need to be careful of the conversion from `u64` to `u32`?
https://github.com/getsentry/snuba/blob/42c564b38a0e85db1cf1307395076656a14a0696/rust_snuba/src/processors/spans.rs#L59

phacops replied:
`start_timestamp_ms` is a `u64` because the column type storing that field is a `UInt64`. Also, it accepts a timestamp in milliseconds.

`timestamp` in the `items` table is a `DateTime` and that's a second precision only timestamp that's backed by a `UInt32`. Anything else than a `u32` is "wasted" because we're inserting into a `UInt32`.

https://clickhouse.com/docs/en/sql-reference/data-types/datetime#speed

kylemumma commented:
changes
* added double write to `insert_int` and `insert_bool` functions
* added all missing attributes

design decisions
* item_type is being hardcoded to 1, is that ok?
* made the design decision to have `timestamp` come from `start_timestamp_ms` is that ok?

phacops commented on rust_snuba/src/processors/eap_items.rs:
```diff
@@ -0,0 +1,390 @@
+                organization_id: from.organization_id,
+                project_id: from.project_id,
+                item_type: 1, // hardcoding "span" as type
+                timestamp: from.start_timestamp_ms as u32,
```
This needs to be divided by 1000 since it's the start timestamp in milliseconds.

phacops reviewed (changes requested):
> item_type is being hardcoded to 1, is that ok?

Yes, that's fine because we'll only have spans coming from that topic.

https://github.com/getsentry/sentry-protos/blob/aea4454db618a2b2e9711fcf86f4356632a3979f/proto/sentry_protos/snuba/v1/request_common.proto#L58

> made the design decision to have timestamp come from start_timestamp_ms is that ok?

Yes, that's fine. `start_timestamp_ms` and `start_timestamp_precise` are the same time in different precisions. Since we're only considering seconds and truncating, both are fine.
